jparest/src/main/java/uk/gov/homeoffice/digital/sas/jparest/service/ResourceApiService.java

```diff
@@ -47,97 +28,44 @@
   }
 
   public List<T> getAllResources(UUID tenantId, Pageable pageable, SpelExpression filter) {
-    var builder = this.entityManager.getCriteriaBuilder();
-    CriteriaQuery<T> query = builder.createQuery(this.entityUtils.getEntityType());
-    Root<T> root = query.from(this.entityUtils.getEntityType());
-
-    var tenantPredicate = builder.equal(root.get(TENANT_ID.getParamName()), tenantId);
-    var filterPredicate = SpelExpressionToPredicateConverter.convert(filter, builder, root);
-    var finalPredicate =
-        filter != null ? builder.and(tenantPredicate, filterPredicate) : tenantPredicate;
-    query.where(finalPredicate);
-
-    CriteriaQuery<T> select = query.select(root);
-    List<Order> orderBy = getOrderCriteria(pageable.getSort(), root, builder);
-    select.orderBy(orderBy);
-
-    EntityGraph<T> entityGraph =
-        this.entityManager.createEntityGraph(this.entityUtils.getEntityType());
-
-    TypedQuery<T> typedQuery = this.entityManager.createQuery(select);
-    typedQuery.setFirstResult((int) pageable.getOffset());
-    typedQuery.setMaxResults(pageable.getPageSize());
-    typedQuery.setHint(QUERY_HINT, entityGraph);
-    return typedQuery.getResultList();
-  }
-
-  private List<Order> getOrderCriteria(Sort sort, Path<?> path, CriteriaBuilder builder) {
-
-    if (sort.isUnsorted()) {
-      return Collections.emptyList();
-    }
-
-    Assert.notNull(path, "Path must not be null!");
-    Assert.notNull(builder, "CriteriaBuilder must not be null!");
-
-    List<Order> orders = new ArrayList<>();
-
-    for (Sort.Order sortOrder : sort) {
-      Order order;
-      if (sortOrder.isAscending()) {
-        order = builder.asc(path.get(sortOrder.getProperty()));
-      } else {
-        order = builder.desc(path.get(sortOrder.getProperty()));
-      }
-      orders.add(order);
-    }
-
-    return orders;
-
+    return repository.findAllByTenantId(tenantId, filter, pageable);
   }
 
-  public List<T> getResource(UUID tenantId, UUID id) {
-    var result = getById(id, null, tenantId);
-    return List.of(result);
+  public T getResource(UUID tenantId, UUID id) {
+    return repository.findByTenantIdAndId(tenantId, id)
+        .orElseThrow(() -> new ResourceNotFoundException(id));
   }
 
-  public List<T> createResource(T payload) {
+  public T createResource(T entity) {
 
     var transactionDefinition = new DefaultTransactionDefinition();
     var transactionStatus = this.transactionManager.getTransaction(transactionDefinition);
     T result;
     try {
-      this.entityValidator.validateAndThrowIfErrorsExist(payload);
-      result = repository.saveAndFlush(payload);
+      this.entityValidator.validateAndThrowIfErrorsExist(entity);
+      result = repository.saveAndFlush(entity);
       transactionManager.commit(transactionStatus);
     } catch (RuntimeException ex) {
       transactionManager.rollback(transactionStatus);
       throw ex;
     }
```

Comment:
This won't be needed if enclosing method is annotated with @Transactional

Reply:
part of the next subtask eahw-2545